libraries/resource_providers/image_streamer/api300/deployment_plan.rb

```diff
@@ -0,0 +1,43 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module ImageStreamer
+    module API300
+      # DeploymentPlan Provider resource methods
+      class DeploymentPlanProvider < ResourceProvider
+        # Generic method to retrieve and return the URI from different image streamer resources using the name of the resource
+        # @param [String, Symbol] i3s_resource_type Type of image streamer resource to be retrieved. e.g., ':GoldenImage'.
+        # @param [String] i3s_resource_name Name of the resource from context.
+        def load_i3s_resource(i3s_resource_type, i3s_resource_name)
```

Comment:
So, @tmiotto @jsmartt , thoughts about this? I'd guess the syntax would become easier to use, but if it seems unecessary just let me know.
Dropping the method would make it so that instead of this:
`@item['goldenImageURI'] = load_i3s_resource(:GoldenImage, @context.golden_image)`
we'd use this:
`          @item['goldenImageURI'] = resource_named(:GoldenImage).find_by(@item.client, @context.golden_image).first['uri'] if @context.golden_image`
Also, in this example, if `@context.golden_image`  contains a valid name, but no golden images exist in the appliance with that name, we'd hit `undefined method []' for nil:NilClass (NoMethodError)`.